### PR TITLE
UIQM-658: Check for duplicate LCCN (010 $a) bib or authority records.

### PR DESCRIPTION
**Pull request by Dmytro-Melnyshyn**
## Purpose
Prevent having bib/authority records with the same 010 $a.

## Description
- validation is added upon saving in `useLccnDuplicationCheck`;
- added permissions to toggle 010 $a validation setting.

## Related PRs

https://github.com/folio-org/ui-marc-authorities/pull/405

## Issues
[UIQM-658](https://folio-org.atlassian.net/browse/UIQM-658)

## Screencast

https://github.com/user-attachments/assets/824ce9df-c0a4-4081-889d-dd4d62a4ecdf

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [ ] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [ ] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/QuickMarcEditor/QuickMarcCreateWrapper.js [hunk 1]
```diff
@@ -102,6 +108,14 @@ const QuickMarcCreateWrapper = ({
   }, [onSave, marcType]);
 
   const onSubmit = useCallback(async (formValues) => {
+    const lccnDuplicationError = await validateLccnDuplication(formValues);
```
Comment (BogdanDenis) [inline src/QuickMarcEditor/QuickMarcCreateWrapper.js:111]: Can we do this valdiation in `QuickMarcEditor` component? We can reduce duplication this way

File: package.json [hunk 2]
```diff
@@ -141,6 +146,37 @@
...
+        "subPermissions": [
+          "ui-quick-marc.settings.lccn-duplicate-check.retrieving",
+          "mod-settings.entries.item.post",
+          "mod-settings.entries.item.put",
+          "mod-settings.global.write.ui-quick-marc.lccn-duplicate-check"
+        ],
+        "visible": true
+      },
+      {
+        "permissionName": "mod-settings.global.read.ui-quick-marc.lccn-duplicate-check",
```
Comment (BogdanDenis) [inline package.json:172]: Are `mod-settings.global.read` and `mod-settings.global.write` prefixes required by mod-settings?

Comment (BogdanDenis) [reply package.json:172]: Also, what is the purpose of these two permissions?

Comment (Dmytro-Melnyshyn) [reply package.json:172]: > Are mod-settings.global.read and mod-settings.global.write prefixes required by mod-settings?

Yes, they are required to have permission to read and write to global settings, creating a new setting in `mod-settings` requires these prefixes and `scope` according to [doc](https://github.com/folio-org/mod-settings/blob/a7239e59328fb515dc6fa65053a105e2c10bc5b5/README.md?plain=1#L35). 

> Also, what is the purpose of these two permissions?

They allow reading and writing this particular setting.

Comment (Dmytro-Melnyshyn) [reply package.json:172]: The `settings/entrie` endpoint requires a `key` and `scope`, the `key` can be anything, the `scope` have to match the scope in `mod-settings.global.read.{scope}`.

Comment (BogdanDenis) [reply package.json:172]: Ok, I understand now. I think the displayName is confusing - if the permission is created just for one scope then let's add it to displayName. Maybe `Settings (MARC): read LCCN duplicate check settings` and `Settings (MARC): write LCCN duplicate check settings`

Comment (Dmytro-Melnyshyn) [reply package.json:162]: I changed the `displayName`, also added the related PR https://github.com/folio-org/ui-marc-authorities/pull/405

**Review by BogdanDenis — APPROVED** [review]

**Review by BogdanDenis — APPROVED** [review]

**Review by Terala-Priyanka — APPROVED** [review]

**Pull request merged**